Src/Base/AMReX_FabArray.H

```diff
@@ -1262,6 +1361,14 @@ FabArray<FAB>::clear ()
         }
     }
     m_fabs_v.clear();
+#ifdef AMREX_USE_GPU
+    The_Pinned_Arena()->free(m_hp_arrays);
```

Comment:
This is relying on the newly changed behavior of CArena that free() is valid even if the pointer wasn't allocated, I take it?

Reply:
No, it does not.  The change in CArena is just to avoid constructing a mutex is the pointer is null.